common/component/azure/eventhubs/eventhubs.go

```diff
@@ -295,15 +298,37 @@ func (aeh *AzureEventHubs) Subscribe(subscribeCtx context.Context, config Subscr
 				}
 			}()
 		}
-	}()
+	}
+	go subscriberLoop()
 
 	// Start the processor
 	go func() {
-		// This is a blocking call that runs until the context is canceled
-		err = processor.Run(subscribeCtx)
-		// Do not log context.Canceled which happens at shutdown
-		if err != nil && !errors.Is(err, context.Canceled) {
-			aeh.logger.Errorf("Error from event processor: %v", err)
+		for {
```

Comment:
```suggestion

	// Start the processor
	go func() {
		for {
	        go subscriberLoop()

```

Reply:
Whoops this came out wrong. The idea was just to move "go subscrierLoop" inside the for loop, so then at the end you don't need to invoke it again. Just for code readability - makes it easier to follow the flow

Reply:
I don't find that easier to read but as you wish.

Reply:
Ah I see what you mean. I thought you wanted me to inline the entire function - which would have been messy. I agree with what you are saying